docs/modules/ROOT/pages/erc1155-uri-storage.adoc

```diff
@@ -0,0 +1,71 @@
+= ERC-1155 URI Storage
+
+The OpenZeppelin xref:erc1155.adoc[ERC-1155] URI Storage extension is needed to manage and store URIs for individual tokens. This extension allows each token to have its own unique URI,
+which can point to metadata about the token, such as images, descriptions, and other attributes.
+This is particularly useful for non-fungible tokens (NFTs) where each token is unique and may have different metadata.
+
+[[usage]]
+== Usage
+
+In order to make an xref:erc1155.adoc[ERC-1155] token with https://docs.rs/openzeppelin-stylus/0.2.0-alpha.2/openzeppelin_stylus/token/erc1155/extensions/uri_storage/index.html[URI Storage] flavour,
+your token should also use https://docs.rs/openzeppelin-stylus/0.2.0-alpha.2/openzeppelin_stylus/token/erc1155/extensions/metadata_uri/index.html[`ERC-1155 Metadata URI`] extension to provide additional URI metadata for each token.
+You need to create a specified contract as follows:
+
+[source,rust]
+----
+use openzeppelin_stylus::token::erc1155::{
+    extensions::{Erc1155MetadataUri, Erc1155UriStorage},
+    Erc1155,
+};
+use stylus_sdk::prelude::{entrypoint, public, sol_storage};
+
+sol_storage! {
+    #[entrypoint]
+    struct Erc1155MetadataUriExample {
+        #[borrow]
+        Erc1155 erc1155;
+        #[borrow]
+        Erc1155MetadataUri metadata_uri;
+        Erc1155UriStorage uri_storage;
+    }
+}
+
+#[public]
+#[inherit(Erc1155, Erc1155MetadataUri)]
+impl Erc1155MetadataUriExample {
+    pub fn uri(&self, token_id: U256) -> String {
+        self.uri_storage.uri(token_id, &self.metadata_uri)
+    }
+
+    #[selector(name = "setTokenURI")]
```

Comment:
I believe `setTokenURI` and `setBaseURI` should not be in the docs as they are not exposed by default (like in Solidity). I believe you should have only `uri`.